src/scenic/core/object_types.py

```diff
@@ -3,666 +3,1760 @@
-		diff = self.position - vec
-		return diff.rotatedBy(-self.heading).y
-
-	def toHeading(self) -> float:
-		return self.heading
+    """The Scenic class ``OrientedPoint``.
+
+    The default mutator for `OrientedPoint` adds Gaussian noise to ``heading``
+    with a standard deviation given by the ``headingStdDev`` property, then
+    applies the mutator for `Point`.
+
+    Properties:
+        yaw (float; dynamic): Yaw of the `OrientedPoint` in radians in the local coordinate system
+          provided by :prop:`parentOrientation`. Default value 0.
+        pitch (float; dynamic): Pitch of the `OrientedPoint` in radians in the local coordinate system
+          provided by :prop:`parentOrientation`. Default value 0.
+        roll (float; dynamic): Roll of the `OrientedPoint` in radians in the local coordinate system
+          provided by :prop:`parentOrientation`. Default value 0.
+        parentOrientation (`Orientation`): The local coordinate system that the `OrientedPoint`'s
+          :prop:`yaw`, :prop:`pitch`, and :prop:`roll` are interpreted in. Default 
+          value is the global coordinate system, where an object is flat in the XY plane,
+          facing North.
+        orientation (`Orientation`; dynamic; final): The orientation of the `OrientedPoint` relative
+          to the global coordinate system. Derived from the :prop:`yaw`, :prop:`pitch`, 
+          :prop:`roll`, and :prop:`parentOrientation` of this `OrientedPoint` and non-overridable.
+        heading (float; dynamic; final): Yaw value of this `OrientedPoint` in the global coordinate
+          system. Derived from :prop:`orientation` and non-overridable.
+        viewAngles (tuple[float,float]): Horizontal and vertical view angles of this `OrientedPoint`
+          in radians. Horizontal view angle can be up to 2π and vertical view angle can be
+          up to π. Values greater than these will be truncated. Default value is (2π, π)
+        orientationStdDev (float): Standard deviation of Gaussian noise to add to this
+          object's Euler angles (yaw, pitch, roll) when mutation is enabled with scale 1.
+          Default value (5π, 0, 0), mutating only the :prop:`yaw` of this `OrientedPoint`.
+    """
+    _scenic_properties = {
+        'yaw': PropertyDefault((), {'dynamic'}, lambda self: 0),
+        'pitch': PropertyDefault((), {'dynamic'}, lambda self: 0),
+        'roll': PropertyDefault((), {'dynamic'}, lambda self: 0),
+        'parentOrientation': Orientation.fromEuler(0, 0, 0),
+
+        'orientation': PropertyDefault(
+            {'yaw', 'pitch', 'roll', 'parentOrientation'},
+            {'dynamic', 'final'},
+            lambda self: (Orientation.fromEuler(self.yaw, self.pitch, self.roll)
+                      * self.parentOrientation)
+        ),
+        'heading': PropertyDefault({'orientation'}, {'dynamic', 'final'},
+            lambda self: self.yaw if alwaysGlobalOrientation(self.parentOrientation) else self.orientation.yaw),
```

Comment:
Better put a comment explaining what's going on here: I remember we discussed it a while back, but it's pretty opaque now!

Reply:
I meant the comment should explain *why* we have this check for whether the parent orientation is the global one: according to the comment `self.orientation.yaw` would be a simpler way to get the same behavior, so we need to explain why we have this more complex code instead.

Reply:
(Looks like this got marked resolved by accident)